adapters/tlsio_mbedtls.c

```diff
@@ -687,19 +719,30 @@ int tlsio_mbedtls_send(CONCRETE_IO_HANDLE tls_io, const void *buffer, size_t siz
         {
             tls_io_instance->send_complete_info.on_send_complete = on_send_complete;
             tls_io_instance->send_complete_info.on_send_complete_callback_context = callback_context;
-            tls_io_instance->send_complete_info.send_complete_count = 0;
-            int res = mbedtls_ssl_write(&tls_io_instance->ssl, buffer, size);
-            if (res != (int)size)
-            {
-                LogError("Unexpected data size returned from  mbedtls_ssl_write %d/%d", res, (int)size);
-                result = MU_FAILURE;
-            }
-            else
+            int out_left = size;
+            result = 0;
+
+            do
             {
-                result = 0;
-            }
+                tls_io_instance->send_complete_info.is_fragmented_req = is_fragmented_send_request(tls_io_instance, out_left);
```

Comment:
does this need to be called every iteration of the loop? pardon my ignorance here but this seems to be able to be set once?

Reply:
yes